# Assignment - Estimation And Confidence Intervals

In [35]:
#Importing Libraries
import numpy as np
from scipy import stats

In [36]:
#Data
#A total of 15 print-heads were randomly selected and tested until failure.
#The durability of each print-head (in millions of characters) was recorded as follows:

data = 1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07, 1.48, 1.20, 1.33, 1.18, 1.22, 1.29


* Q.  Build 99% Confidence Interval Using Sample Standard Deviation

In [37]:
#Assuming the sample is representative of the population, construct a 99% confidence interval for the mean number of characters printed before the print-head fails using the sample standard deviation.
#Explain the steps you take and the rationale behind using the t-distribution for this task.

In [38]:
data

(1.13,
 1.55,
 1.43,
 0.92,
 1.25,
 1.36,
 1.32,
 0.85,
 1.07,
 1.48,
 1.2,
 1.33,
 1.18,
 1.22,
 1.29)

In [39]:
len(data)

15

In [40]:
#calculating the sample mean and sample standard deviation

mean = np.mean(data)
print(mean)
std = np.std(data,ddof=1)
print(data)

1.2386666666666666
(1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07, 1.48, 1.2, 1.33, 1.18, 1.22, 1.29)


In [41]:
#finding the critical t-value.
#Since we have a 99% confidence interval and 14 degrees of freedom (n-1 = 15-1 = 14)

t_critical = stats.t.ppf(q=0.99,df=14)
print(t_critical)

2.624494067560231


In [42]:
#Calculating margin of error

margin_of_error = t_critical * (std/np.sqrt(15))
margin_of_error

0.1308960216979201

In [43]:
#Constructing confidence interval

lower_bound = mean - margin_of_error
upper_bound = mean + margin_of_error

confidence_interval = (lower_bound,upper_bound)
print(confidence_interval)

(1.1077706449687466, 1.3695626883645866)


99% confidence interval for the mean number of characters printed before the print-head fails using the sample standard deviation is ,  (1.1077706449687466, 1.3695626883645866)

Conclusion :

We use the t-distribution because the population standard deviation is unknown, and the sample size is small (n < 30). The t-distribution is a more robust distribution than the normal distribution when dealing with small samples and unknown population standard deviations

* Q. Build 99% Confidence Interval Using Known Population Standard Deviation

In [44]:
#If it were known that the population standard deviation is 0.2 million characters,
#construct a 99% confidence interval for the mean number of characters printed before failure.

In [45]:
#Calculating the sample mean
pop_mean = np.mean(data)
pop_mean

1.2386666666666666

In [46]:
#Known population standard deviation
pop_std = 0.2
pop_std

0.2

In [47]:
#Calculating the critical z-value
z_critical = stats.norm.ppf(0.995)
z_critical

2.5758293035489004

In [48]:
#Calculating margin of error
margin_of_error = z_critical * (pop_std/np.sqrt(15))
margin_of_error

0.13301525327090588

In [49]:
#Constructing confidence interval

lower_bound = pop_mean - margin_of_error
upper_bound = pop_mean + margin_of_error

confidence_interval = (lower_bound,upper_bound)
confidence_interval

(1.1056514133957607, 1.3716819199375725)

99% confidence interval for the mean number of characters printed before failure is , (1.1056514133957607, 1.3716819199375725)

Conclusion :

We use the z-distribution (standard normal distribution) because the population standard deviation is known.

The critical z-value is calculated using the stats.norm.ppf() function from the scipy.stats library. The argument 0.995 corresponds to the 99.5th percentile of the standard normal distribution, which is used for a two-tailed 99% confidence interval.